components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

```diff
@@ -146,7 +146,13 @@ fun String.sanitizeURL(): String {
  * For example for an input of "/../../../../../../directory/file.txt" you will get "file.txt"
  */
 fun String.sanitizeFileName(): String {
-    return this.substringAfterLast(File.separatorChar)
+    val file = File(this.substringAfterLast(File.separatorChar))
+    // Remove unwanted subsequent dots in the file name.
+    return if (file.extension.trim().isNotEmpty()) {
+        file.name.replace("\\.\\.+".toRegex(), ".")
+    } else {
+        file.name.replace("\\.\\.?+".toRegex(), "")
```

Comment:
Adding as we talked about this and I will forget :). Here we should just be able to do `replace(".", "")` and then we also don't need the `removePrefix` below.